test/unit/transform/eia176_test.py

```diff
@@ -0,0 +1,207 @@
+    "1010",
+    "2022",
+    VOLUME_1,
+    "[10.1]",
+    "residential sales volume",
+]
+
+ID_2 = "17635017NM"
+VOLUME_2 = 532842.0
+COMPANY_2 = [
+    "new mexico",
+    "vl",
+    "west texas gas inc",
+    ID_2,
+    "1010",
+    "2022",
+    VOLUME_2,
+    "[10.1]",
+    "residential sales volume",
+]
+
+NM_VOLUME = VOLUME_1 + VOLUME_2
+NM_AGGREGATE = [
+    "new mexico",
+    "vl",
+    "total of all companies",
+    # Aggregates appear to reuse an arbitrary company ID
+    ID_1,
+    "1010",
+    "2022",
+    NM_VOLUME,
+    "[10.1]",
+    "residential sales volume",
+]
+
+ID_3 = "17635017TX"
+VOLUME_3 = 1.0
+COMPANY_3 = [
+    "texas",
+    "vl",
+    "west texas gas inc",
+    ID_3,
+    "1010",
+    "2022",
+    VOLUME_3,
+    "[10.1]",
+    "residential sales volume",
+]
+
+TX_VOLUME = VOLUME_3
+TX_AGGREGATE = [
+    "texas",
+    "vl",
+    "total of all companies",
+    # Aggregates appear to reuse an arbitrary company ID
+    ID_3,
+    "1010",
+    "2022",
+    VOLUME_3,
+    "[10.1]",
+    "residential sales volume",
+]
+
+US_VOLUME = NM_VOLUME + TX_VOLUME
+US_AGGREGATE = [
+    "u.s. total",
+    "vl",
+    "total of all companies",
+    # Aggregates appear to reuse an arbitrary company ID
+    ID_1,
+    "1010",
+    "2022",
+    US_VOLUME,
+    "[10.1]",
+    "residential sales volume",
+]
+
+ID_4 = "4"
+VOLUME_4 = 4.0
+COMPANY_4 = [
+    "alaska",
+    "vl",
+    "alaska gas inc",
+    ID_4,
+    "1020",
+    "2022",
+    VOLUME_4,
+    "[10.2]",
+    "some other volume",
+]
+
+DROP_COLS = ["itemsort", "item", "atype", "line", "company"]
+
+
+@fixture
+def df():
+    df = pd.DataFrame(columns=COLUMN_NAMES)
+    df.loc[0] = COMPANY_1
+    df.loc[1] = COMPANY_2
+    df.loc[2] = NM_AGGREGATE
+    df.loc[3] = COMPANY_3
+    df.loc[4] = TX_AGGREGATE
+    df.loc[5] = US_AGGREGATE
+    df.loc[6] = COMPANY_4
+    df = df.set_index(["area", "company"])
+    return df
+
+
+def test_core_eia176__data(df):
+    eav_model = df.loc[
+        [
+            ("new mexico", "new mexico gas company"),
+            ("new mexico", "total of all companies"),
+        ]
+    ].reset_index()
+
+    wide_company, wide_aggregate = _core_eia176__data(eav_model)
+    assert wide_company.shape == (1, 4)
+
+    company_row = wide_company.loc[0]
+    assert list(company_row.index) == ["report_year", "area", "id", "1010_vl"]
+    assert list(company_row.values) == [2022, "new mexico", ID_1, VOLUME_1]
+
+    assert wide_aggregate.shape == (1, 3)
+    aggregate_row = wide_aggregate.loc[0]
+    assert list(aggregate_row.index) == ["report_year", "area", "1010_vl"]
+    assert list(aggregate_row.values) == [2022, "new mexico", NM_VOLUME]
+
+
+def test_get_wide_table(df):
+    long_table = df.loc[
+        [
+            ("new mexico", "new mexico gas company"),
+            ("new mexico", "west texas gas inc"),
+            # Row measuring a different variable to test filling NAs
+            ("alaska", "alaska gas inc"),
+        ]
+    ].reset_index()
```

Comment:
A little verbose here with the initial setting of label indices and resetting here, but I think it's worth it to communicate what's going into this test and the properties it's checking.

Reply:
That's just how pandas is IME. Endless setting and resetting of indices. One must imagine Sisyphus happy etc.